src/Cns.ts

```diff
@@ -227,6 +227,18 @@ export default class Cns extends NamingService {
     return !isNullAddress(data.owner);
   }
 
+  async getTokenUri(tokenId: string): Promise<string> {
+    try {
+      const [tokenUri] = await this.readerContract.call('tokenURI', [tokenId]);
+      return tokenUri;
+    } catch (error) {
```

Comment:
@DancingAxolotl It worth checking whether this error means that domain is not registered. We could get an error just in  case of connection issues. In this case we need to throw original error to the client.